internal/services/cognitive/azure_ai_services_resource.go

```diff
@@ -0,0 +1,757 @@
+			}
+			if customMangedKey != nil {
+				props.Properties.Encryption = customMangedKey
+			}
+
+			expandIdentity, err := identity.ExpandSystemAndUserAssignedMapFromModel(model.Identity)
+			if err != nil {
+				return fmt.Errorf("expanding `identity`: %+v", err)
+			}
+			props.Identity = expandIdentity
+
+			future, err := client.AccountsUpdate(ctx, *id, props)
+			if err != nil {
+				return fmt.Errorf("updating %s: %+v", id, err)
+			}
+
+			if err := future.Poller.PollUntilDone(ctx); err != nil {
+				return fmt.Errorf("waiting for updating of %s: %+v", id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (AzureAIServicesResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 180 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Cognitive.AccountsClient
+
+			id, err := cognitiveservicesaccounts.ParseAccountID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			log.Printf("[DEBUG] Retrieving %s..", *id)
+			account, err := client.AccountsGet(ctx, *id)
+			if err != nil || account.Model == nil || account.Model.Location == nil {
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			deletedAzureAIServicesId := cognitiveservicesaccounts.NewDeletedAccountID(id.SubscriptionId, *account.Model.Location, id.ResourceGroupName, id.AccountName)
+			if err != nil {
+				return err
+			}
+
+			log.Printf("[DEBUG] Deleting %s..", *id)
+			if err := client.AccountsDeleteThenPoll(ctx, *id); err != nil {
+				return fmt.Errorf("deleting %s: %+v", *id, err)
+			}
+			if metadata.Client.Features.CognitiveAccount.PurgeSoftDeleteOnDestroy {
+				log.Printf("[DEBUG] Purging %s..", *id)
+				if err := client.DeletedAccountsPurgeThenPoll(ctx, deletedAzureAIServicesId); err != nil {
+					return fmt.Errorf("purging %s: %+v", *id, err)
+				}
+			} else {
+				log.Printf("[DEBUG] Skipping Purge of %s", *id)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (AzureAIServicesResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return cognitiveservicesaccounts.ValidateAccountID
+}
+
+func expandAzureAIServicesCustomerManagedKey(input []AzureAIServicesCustomerManagedKey) (*cognitiveservicesaccounts.Encryption, error) {
+	if len(input) == 0 {
+		return nil, nil
+	}
+
+	v := input[0]
+	keySource := cognitiveservicesaccounts.KeySourceMicrosoftPointKeyVault
+
+	var identityClientId string
+	if value := v.IdentityClientID; value != "" {
+		identityClientId = value
+	}
+
+	encryption := &cognitiveservicesaccounts.Encryption{
+		KeySource: &keySource,
+		KeyVaultProperties: &cognitiveservicesaccounts.KeyVaultProperties{
+			IdentityClientId: utils.String(identityClientId),
+		},
+	}
+
+	if v.KeyVaultKeyID != "" {
+		keyId, err := keyVaultParse.ParseOptionallyVersionedNestedItemID(v.KeyVaultKeyID)
+		if err != nil {
+			return nil, fmt.Errorf(" Failed to parse '%s' as Key Vault key ID", keySource)
```

Comment:
As mentioned above, this doesn't need wrapping
```suggestion
			return nil, err
```